nixos/modules/services/mail/dovecot.nix

```diff
@@ -131,6 +247,60 @@ let
       };
     };
   };
+
+  imapsieveMailboxes = {
+    options = {
+      name = mkOption {
+        type = types.str;
+        example = "Junk";
+        description = ''
+          Name of a mailbox for which administrator scripts are configured.
+          This setting supports wildcards with a syntax compatible with the
+          IMAP LIST command, meaning that this setting can apply to multiple or
+          even all ("*") mailboxes.
+        '';
+      };
+      beforeScript = mkOption {
+        type = types.nullOr types.path;
```

Comment:
What is the reason to ask for a path to a sieve script instead of an actual string containing the sieve script here?

Reply:
> On the other hand if this option was a string and I prefer to store my sieve scripts in files I would have to use fileContents to load the data from the file, only for the module to write that content to a different file which seems a waste

Yes, but using a string is standard… and it's not like there's not a lot of copying going on during builds ;)

Reply:
@wmertens I know. I guess I just like `writeText` more than `fileContents`.

I have considered supporting both `strings` and `path` by making the value an `attrset` like `environment.etc` or `services.rspamd.locals` but a `listOf attrset` is not as nice to work with as the named submodules like `environment.etc` and `services.rspamd.locals`.

Reply:
How about having both text and path attributes? Or, if a Sieve script can never be a single line starting with `/`, autodetect?

Reply:
@wmertens I looked at the Sieve RFC and from that it seems that while a sieve script can start with `/` it has to start with `/*` in that case which is obviously not a valid store path. So I am working on an update for this PR the resolves merge conflicts and supports having both text and path attributes for Sieve scripts.